### ISSUE TITLE
Report generation takes longer with larger projects

### ISSUE THREAD

**Issue by ominestre** [ominestre/rotten-deps#35]
**Is your feature request related to a problem? Please describe.**
When generating a report on a project with 85 dependencies the runtime was ~20-30 seconds.

**Describe the solution you'd like**
Since this can appear as if the command is hanging a simple message describing that this process takes longer on larger projects should be sufficient.

**Comment by ominestre** [comment]
https://www.npmjs.com/package/cli-progress

Would be another solution. `maestro` https://github.com/ominestre/rotten-deps/blob/master/src/bin/rotten-deps.ts#L47 is what orchestrates the entire process for generating the report so it might not be much effort to work in a progress bar.

I would also throw an additional requirement that if we go with progress bars that we add a `--no-progress` flag which squelches this.

**Comment by ominestre** [comment]
f8c672488acb1f4c95d53efc53d9963fa14eca63 adds hooks into the report generator that could be used for tracking progress. This is the long pull right now because of the individual detail requests on each outdated dependency in order to fetch version release dates.

To early to tell but I did play around with `cli-progress` but the approach I took wasn't displaying the bars. I'm going to come at this fresh later on.
